tests/php/Forms/GridField/GridFieldExportButtonTest.php

```diff
@@ -55,11 +55,10 @@ public function testCanView()
         $config = GridFieldConfig::create()->addComponent(new GridFieldExportButton());
         $gridField = new GridField('testfield', 'testfield', $list, $config);
 
-        $csvReader = Reader::createFromString($button->generateExportFileData($gridField));
-        $bom = $csvReader->getInputBOM();
+        $csvReader = $this->createReader($button->generateExportFileData($gridField));
 
         $this->assertEquals(
-            "$bom\"My Name\"\r\n",
```

Comment:
Why are we removing the bom? Isn't the point that this proves an input with BOM gives output without it? So it's fundamentally part of the test?

though this doesn't need to be part of every assertion, it should be in one, I guess?

Reply:
Fair call, we should revise. As part of this I think we need to amend the build matrix to include a build with league/csv 8. Perhaps our 7.1 build should be reconfigured to include --prefer-lowest?

Reply:
OK I've done this and it's exposed that we're currently failing tests with league/csv 8 in place. So that needs to be fixed.

Reply:
Personally, I'd be happy to drop league/csv ^7. Our obligation is to our public APIs, not other modules, but I can see that there is value to wider support and developers using league/csv will have used 7 because we are even if they had wanted to use 8.

Reply:
The issue is one of upgradeability: if we drop support for a major release line of a dependency which a project has relied on, then they need to do a major upgrade of that package to do a minor upgrade of ours.

So adding support for new majors but not deleting old ones should be the norm.

I'll admit that we did drop support for a major dependency in that case of PHP 5.6; but I see that as the exception rather than the norm, and relates to the substantially increased level of risk that someone has using an EOL version of PHP rather than an actively supported one - in short, not being able to upgrade PHP is a much bigger issue than not being able to upgrade Silverstripe in such a case ;-)

We also had years where 5.x and 7.x were supported in parallel.

I think we could probably use those reasons to justify dropping support for past-EOL versions of symfony. But I think for league continuing to support 8.x and 9.x is the right move. I should write this up as an RFC really ;-)

Reply:
FYI I assumed that you meant "Personally, I'd be happy to drop league/csv ^8", not 7, as we're not currently supporting 7.

Reply:
Restored the original state of the tests, and enforced output of BOM from the `Reader` in the tests (as `league/csv ^9` strips it by default.) Tests are green against both versions of the package now.